akka-projection-core/src/main/scala/akka/projection/scaladsl/SourceProvider.scala

```diff
@@ -26,6 +27,42 @@ trait SourceProvider[Offset, Envelope] {
 
 }
 
+/**
+ * By default, a `SourceProvider` uses the stored offset when starting the Projection. This offset can be adjusted
+ * by decorating the `SourceProvider` with this class and defining the `adjustStartOffset` function.
+ *
+ * @param loadFromOffsetStore whether the offset should be loaded from the offset store or not, loaded offset
+ *                            is passed as the parameter to the `startOffset` function
+ * @param adjustStartOffset function from loaded offset (if any) to the adjusted offset
+ * @param delegate the original `SourceProvider`
+ */
+final class CustomStartOffsetSourceProvider[Offset, Envelope](
+    loadFromOffsetStore: Boolean,
+    adjustStartOffset: Option[Offset] => Future[Option[Offset]],
```

Comment:
The idea is that the user would decorate the ordinary SourceProvider by wrapping it with this